api/environments/component_handler.go

```diff
@@ -126,6 +131,64 @@ func (eh EnvironmentHandler) RestartComponent(appName, envName, componentName st
 	return nil
 }
 
+// RestartComponentAuxiliaryResource Restarts a component's auxiliary resource
+func (eh EnvironmentHandler) RestartComponentAuxiliaryResource(appName, envName, componentName, auxType string) error {
+	log.Infof("Restarting auxiliary resource %s for component %s, %s", auxType, componentName, appName)
+
+	deploySummary, err := eh.deployHandler.GetLatestDeploymentForApplicationEnvironment(appName, envName)
+	if err != nil {
+		return err
+	}
+
+	deploymentDto, err := eh.deployHandler.GetDeploymentWithName(appName, deploySummary.Name)
+	if err != nil {
+		return err
+	}
+
+	componentDto := deploymentDto.GetComponentByName(componentName)
+	if componentDto == nil {
+		return environmentModels.NonExistingComponent(appName, componentName)
+	}
+
+	auxResourceDto := componentDto.GetAuxiliaryResourceByType(auxType)
+	if auxResourceDto == nil {
+		return environmentModels.NonExistingComponentAuxiliaryType(appName, componentName, auxType)
+	}
+
+	if auxResourceDto.Deployment == nil || auxResourceDto.Deployment.Status != deploymentModels.ConsistentComponent.String() {
+		currentStatus := deploymentModels.ComponentReconciling.String()
+		if auxResourceDto.Deployment != nil {
+			currentStatus = auxResourceDto.Deployment.Status
+		}
+		return environmentModels.CannotRestartComponent(appName, componentName, currentStatus)
+	}
+
+	selector := labelselector.ForAuxiliaryResource(appName, componentName, auxType).String()
+	envNs := operatorUtils.GetEnvironmentNamespace(appName, envName)
+	deployments, err := eh.client.AppsV1().Deployments(envNs).List(context.TODO(), metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return err
+	}
+	if len(deployments.Items) == 0 {
+		return environmentModels.CannotRestartComponent(appName, componentName, deploymentModels.ComponentReconciling.String())
+	}
+
+	return eh.patchDeploymentForRestart(&deployments.Items[0])
+}
+
+func (eh EnvironmentHandler) patchDeploymentForRestart(deployment *appsv1.Deployment) error {
+	if deployment.Spec.Template.Annotations == nil {
+		deployment.Spec.Template.Annotations = make(map[string]string)
+	}
+
+	deployment.Spec.Template.Annotations[restartedAtAnnotation] = radixutils.FormatTimestamp(time.Now())
+	if _, err := eh.client.AppsV1().Deployments(deployment.Namespace).Update(context.TODO(), deployment, metav1.UpdateOptions{}); err != nil {
```

Comment:
Should we use the method like in the `radix-operator` to update status:
`retry.RetryOnConflict(retry.DefaultRetry, func() error {...}`

Reply:
Agree. Fixed.